### ISSUE TITLE
make error for stereomapper

### ISSUE THREAD

**Issue by dejunzhang** [amiltonwong/libviso2#1]
Hi, wong,
when i run make in stereomapper: there are some errors appears, like below:
../libviso2/src/visualodometry.h:31:7: note: VisualOdometry::VisualOdometry(const VisualOdometry&)
 class VisualOdometry
       ^
../libviso2/src/visualodometry.h:31:7: note:   candidate expects 1 argument, 0 provided
../libviso2/src/visualodometry.h:31:7: note: VisualOdometry::VisualOdometry(VisualOdometry&&)
../libviso2/src/visualodometry.h:31:7: note:   candidate expects 1 argument, 0 provided
visualodometrythread.cpp: In member function ‘virtual void VisualOdometryThread::run()’:
visualodometrythread.cpp:61:82: error: no matching function for call to ‘Matcher::pushBack(unsigned char_&, unsigned char_&, int&, int&, int&, bool)’
     matcher->pushBack(simg->I1,simg->I2,simg->width,simg->height,simg->step,false);
                                                                                  ^
visualodometrythread.cpp:61:82: note: candidates are:
In file included from visualodometrythread.h:11:0,
                 from visualodometrythread.cpp:1:
../libviso2/src/matcher.h:111:8: note: void Matcher::pushBack(uint8_t_, uint8_t_, int32_t_, bool)
   void pushBack (uint8_t *I1,uint8_t_ I2,int32_t\* dims,const bool replace);
        ^
../libviso2/src/matcher.h:111:8: note:   candidate expects 4 arguments, 6 provided
../libviso2/src/matcher.h:117:8: note: void Matcher::pushBack(uint8_t_, int32_t_, bool)
   void pushBack (uint8_t _I1,int32_t_ dims,const bool replace) { pushBack(I1,0,dims,replace); }
        ^
../libviso2/src/matcher.h:117:8: note:   candidate expects 3 arguments, 6 provided
visualodometrythread.cpp:76:11: error: ‘class VisualOdometry’ has no member named ‘setCalibration’
       vo->setCalibration(f,cu,cv,b);
           ^
visualodometrythread.cpp:77:11: error: ‘class VisualOdometry’ has no member named ‘update’
       vo->update(matches,timeDiff(time_curr,time_prev),true,record_raw_odometry);
           ^
visualodometrythread.cpp:79:40: error: conversion from ‘cvm::rmatrix {aka cvm::basic_rmatrix<double>}’ to non-scalar type ‘Matrix’ requested
       Matrix H = vo->getTransformation();
                                        ^
make: **\* [visualodometrythread.o] Error 1

It means that there are some interfaces missing.  Could you please kindly tell me where can i found them or how to write them myself? thank you very much.

**Comment by AndreiBarsan** [comment]
I'm getting the same problem (after tweaking a lot more stuff, like ensuring c++11 support was enabled, etc.).

It seems that StereoMatcher has been programmed against a different (older?) version of libviso2. Interestingly enough, I've checked out libviso (the "1.0" version), and the matcher's interface seems to match, but `VisualOdometry`s is still wrong.

I would be very grateful to know what changes @amiltonwong made to the code in order to get it to build and run successfully (if it was at all possible).

**Comment by willSapgreen** [comment]
Based on my observation,
you can find VisualOdometry is declared/defined in the latest viso.h/.cpp in libviso2
For some VisualOdometry's functions( update, setCalibration, getTransformation, getInliers )
you need to modify either visualodometrythread.cpp or viso.h/.cpp to make work, which I am working on now.
Please let me know if any suggestion or misunderstanding.

**Comment by AndreiBarsan** [comment]
@willSapgreen Yeah, I may work on getting this to work in the future, but I don't know for sure at the moment. Please do share any progress you make getting StereoMapper to work! :)

**Comment by ktxu1224** [comment]
Hi, I am meeting the same error while using the viso.cpp. Would you like to tell me if you have solved the problem mentioned above? @willSapgreen @AndreiBarsan   Thanks a lot
